app/gui/view/src/project.rs

```diff
@@ -523,20 +517,79 @@ impl View {
+                    (searcher.as_ref()?.input == *node_id).then(input_change)
+                }
+            ).on_change();
+            input_change <- searcher_input_change_opt.unwrap();
+            // We wait with processing an updated input to avoid unnecessary refreshes during
+            // typing. This is done by delaying the input change by a short amount of time.
+            input_change_delay.restart <+ input_change.constant(INPUT_CHANGE_DELAY_MS);
+
+            // When accepting the input, we need to make sure we correctly process any outstanding
+            // key presses. If we don't, we may end up with a stale selection. But we do not want
+            // to refresh the searcher if we don't have outstanding key presses, as this will
+            // cause the searcher to lose the selected entry and instead select the default entry.
+
+            let needs_refresh = input_change_delay.is_running;
+            // We have retained key presses that we need to process.
+            update_with_refresh <- committed_in_searcher.gate(&needs_refresh);
+            // No key presses retained, accept the selection as is.
+            update_without_refresh <- committed_in_searcher.gate_not(&needs_refresh);
+
+            on_update_with_refresh <- update_with_refresh.constant(());
+            // We only need to cancel the delay if the timer is running, that is, if we have
+            // unprocessed changes and need to refresh.
+            input_change_delay.cancel <+ on_update_with_refresh;
+
+            update_input <- any(&input_change_delay.on_expired, &on_update_with_refresh);
+            input_change_and_searcher <-
+                all_with(&input_change, &frp.searcher, |c, s| (c.clone(), *s));
+            updated_input <- input_change_and_searcher.sample(&update_input);
+            input_changed <- updated_input.filter_map(|((node_id, expr, selections), searcher)| {
+                let input_change = || (expr.clone_ref(), selections.clone());
+                (searcher.as_ref()?.input == *node_id).then(input_change)
+            });
+            frp.source.searcher_input_changed <+ input_changed;
+
+            // Since the input was updated, the selected item might have been replaced, so we need
+            // discard it and instead use the currently selected item. Note that this will be the
+            // default item, and we loose any selection that was made. But that is fine, as the
+            // user was typing and a change to the selection was to be expected.
+            frp.source.editing_committed <+ on_update_with_refresh.map3(
+                &committed_in_searcher,&grid.active,
+                |_, (node_id, _), &entry| Some((*node_id, entry?.as_entry_id()))).unwrap();
+
+            // If we have no outstanding key presses, we can accept the selection as is.
+            frp.source.editing_committed <+ committed_in_searcher.sample(&update_without_refresh);
```

Comment:
This could belong to the next subsection (as committing edit closed the searcher).

If, if such separation is more readable to you, let's rename this section to "handling input and committing edit".